app/factories/bulkrax/object_factory.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require_dependency Bulkrax::Engine.root.join('app', 'factories', 'bulkrax', 'object_factory')
+
+Bulkrax::ObjectFactory.class_eval do     # rubocop:disable Metrics/ParameterLists
+    # rubocop:disable Metrics/ParameterLists
+    def initialize(attributes:, source_identifier_value:, work_identifier:, collection_field_mapping:, replace_files: false, user: nil, klass: nil, update_files: false, importer: nil)
+      @attributes = ActiveSupport::HashWithIndifferentAccess.new(attributes)
+      @replace_files = replace_files
+      @update_files = update_files
+      @user = user || User.batch_user
+      @work_identifier = work_identifier
+      @collection_field_mapping = collection_field_mapping
+      @source_identifier_value = source_identifier_value
+      @klass = klass || Bulkrax.default_work_type.constantize
+      @importer = importer
+    end
+
+  # Regardless of what the Parser gives us, these are the properties we are prepared to accept.
+  def permitted_attributes
+    klass.properties.keys.map(&:to_sym) + %i[id edit_users edit_groups read_groups visibility work_members_attributes admin_set_id member_of_collections_attributes pbcore_xml skip_file_upload_validation bulkrax_importer_id]
+  end
+
+  def environment(attrs)
+    Hyrax::Actors::Environment.new(object, Ability.new(@user), attrs, @importer)
+  end
+end
```

Comment:
this is allowing for the passing of the importer "flag" to the actor stack.  The need for the actor stack is for the queued indexing extent implementation. That's not in Bulkrax or Hyrax yet, because we architected it here in GBH to solve the indexing speed issue.

Reply:
👍